src/test/test_addr.c

```diff
@@ -1671,6 +1671,128 @@ test_addr_octal(void *arg)
   ;
 }
 
+#define Get_IPv4(test_addr, str) STMT_BEGIN                     \
+    test_addr = tor_malloc(sizeof(tor_addr_t));                 \
+    test_addr->family = AF_INET;                                \
+    tor_inet_aton(str, &test_addr->addr.in_addr);               \
+  STMT_END;
+
+#define Get_IPv6(test_addr, str) STMT_BEGIN                     \
+    test_addr = tor_malloc(sizeof(tor_addr_t));                 \
+    test_addr->family = AF_INET6;                               \
+    tor_inet_pton(AF_INET6, str, &test_addr->addr.in6_addr);    \
+  STMT_END;
+
+#define Get_AF_UNIX(test_addr) STMT_BEGIN                       \
+    test_addr = tor_malloc_zero(sizeof(tor_addr_t));            \
+    test_addr->family = AF_UNIX;                                \
+  STMT_END;
+
+#define Get_AF_UNSPEC(test_addr) STMT_BEGIN                     \
+    test_addr = tor_malloc_zero(sizeof(tor_addr_t));            \
+    test_addr->family = AF_UNSPEC;                              \
+  STMT_END;
+
+#define TEST_ADDR_VALIDITY(a, lis, rv) STMT_BEGIN               \
+    tor_assert(a);                                              \
+    tt_int_op(tor_addr_is_valid(a, lis), OP_EQ, rv);            \
+  STMT_END;
+
+static void
+test_addr_is_valid(void *arg)
+{
+  (void)arg;
+  tor_addr_t *test_addr;
+
+  /* Tests for IPv4 addresses. */
+
+  /* Test for null IPv4 address. */
+  Get_IPv4(test_addr, "0.0.0.0");
+  TEST_ADDR_VALIDITY(test_addr, 0, 0);
+  TEST_ADDR_VALIDITY(test_addr, 1, 1);
+  tor_free(test_addr);
+
+  /* Test for non-null IPv4 address. */
+  Get_IPv4(test_addr, "123.98.45.1");
+  TEST_ADDR_VALIDITY(test_addr, 0, 1);
+  TEST_ADDR_VALIDITY(test_addr, 1, 1);
+  tor_free(test_addr);
+
+  /* Tests for IPv6 addresses. */
+
+  /* Test for null IPv6 address. */
+  Get_IPv6(test_addr, "::");
+  TEST_ADDR_VALIDITY(test_addr, 0, 0);
+  TEST_ADDR_VALIDITY(test_addr, 1, 1);
+  tor_free(test_addr);
+
+  /* Test for non-null IPv6 address. */
+  Get_IPv6(test_addr, "2000::1a00::1000:fc098");
```

Comment:
This is not an IPv6 address, it has two "::".

Looks like Windows and macOS/Linux handle this failure differently.
That shouldn't happen, I've opened a ticket to fix it:
https://trac.torproject.org/projects/tor/ticket/33768